arviz/utils/xarray_utils.py

```diff
@@ -1,454 +1,246 @@
+            pymc3 trace: Automatically extracts data
+            xarray.Dataset: adds to InferenceData as only group
+            dict: creates an xarray dataset as the only group
+            numpy array: creates an xarray dataset as the only group, gives the
+                         array an arbitrary name
+    group : str
+        If `obj` is a dict or numpy array, assigns the resulting xarray
+        dataset to this group.
     coords : dict[str, iterable]
         A dictionary containing the values that are used as index. The key
         is the name of the dimension, the values are the index values.
-    dims : dict[str, Tuple(str)]
-        A mapping from pymc3 variables to a tuple corresponding to
-        the shape of the variable, where the elements of the tuples are
-        the names of the coordinate dimensions.
-    chains : int or None
-        The number of chains sampled from the posterior, only necessary for
-        converting dicts.
+    dims : dict[str, List(str)]
+        A mapping from variables to a list of coordinate names for the variable
 
     Returns
     -------
-    InferenceData
-        This wraps a netCDF datset representing those groups available to the object.
-        The coordinates are those passed in and ('chain', 'draw')
+    xarray.Dataset
     """
-    if isinstance(obj, InferenceData):
-        return obj
+    inference_data = convert_to_inference_data(obj, group=group, coords=coords, dims=dims)
+    dataset = getattr(inference_data, group, None)
+    if dataset is None:
+        raise ValueError('Can not extract {group}! See {filename} for other '
+                         'conversion utilities.'.format(group=group, filename='__file__'))
+    return dataset
+
+
+class requires: # pylint: disable=invalid-name
+    """Decorator to return None if an object does not have the required attribute"""
+    def __init__(self, *props):
+        self.props = props
+
+    def __call__(self, func):
+        def wrapped(cls, *args, **kwargs):
+            for prop in self.props:
+                if getattr(cls, prop) is None:
+                    return None
+            return func(cls, *args, **kwargs)
+        return wrapped
+
+
+def numpy_to_data_array(ary, *_, var_name='data', coords=None, dims=None):
+    default_dims = ['chain', 'draw']
+    if dims is None:
+        can_squeeze = False
+        if len(ary.shape) < 3:
+            ary = np.atleast_3d(ary)
+            can_squeeze = True # added a dimension, might remove it too
+        n_chains, n_samples, *shape = ary.shape
+        if n_chains > n_samples:
+            warnings.warn('More chains ({n_chains}) than draws ({n_samples}). '
+                          'Passed array should have shape (chains, draws, *shape)'.format(
+                              n_chains=n_chains, n_samples=n_samples), SyntaxWarning)
+
+        coords = {}
+        dims = default_dims
+        if can_squeeze and len(shape) == 1 and shape[0] == 1:
```

Comment:
Perhaps a comment here on what case this handles would be good. It's not obvious, at least to me what this checks for and what problem it solves